pymc3/variational/advi.py

```diff
@@ -294,3 +296,53 @@ def adagrad(grad, param, learning_rate, epsilon, n):
     updates[param] = param - (-learning_rate * grad /
                               theano.tensor.sqrt(accu_sum + epsilon))
     return updates
+
+def sample_vp(draws, model, vparams, random_seed=20090425):
```

Comment:
I think you can use `model=None` and then try to get it from the `modelcontext` to enable the `with` context manager.